### ISSUE TITLE
How to decode the external trigger events with a live event camera？

### ISSUE THREAD

**Issue by NetaPanda** [ros-event-camera/event_camera_py#18]
Hi, I am trying to use event_camera_py to decode the events directly from my event camera rather than from the ROS bags. I am using an EVK4 with metavision_driver and I use ROS2 humble.

Below is a simple subscriber I wrote:

```
import rclpy
from rclpy.node import Node
from rclpy.serialization import deserialize_message
import numpy as np
from event_camera_msgs.msg import EventPacket
from event_camera_py import Decoder
from event_camera_py import UniqueDecoder

class TriggeredEventSegmentsPublisher(Node):
    def __init__(self):
        super().__init__('triggered_event_segments_publisher')
        EVENT_TOPIC =  '/event_camera/events' 
        self.decoder = Decoder()
        qsize = 1000
        # some quality of service settings that match the event_camera_py package
        qos_policy = rclpy.qos.QoSProfile(reliability=rclpy.qos.ReliabilityPolicy.BEST_EFFORT,
                                          history=rclpy.qos.HistoryPolicy.KEEP_LAST,
                                          durability=rclpy.qos.DurabilityPolicy.VOLATILE,
                                          depth=qsize)
        self.subscription = self.create_subscription(
            EventPacket,
            EVENT_TOPIC,
            self.listener_callback,
            qos_profile=qos_policy)

        self.subscription  # prevent unused variable warning

        self.event_queue = None


    # function once got an event packet from the topic
    def listener_callback(self, msg):
        self.get_logger().info('I heard a event packet')
        # decode the msg
        self.decoder.decode(msg)
        cd_events = self.decoder.get_cd_events()
        if self.event_queue is None:
            self.event_queue = cd_events
        else:
            self.event_queue = np.hstack((self.event_queue, cd_events))
        print(cd_events)
        trig_events = self.decoder.get_ext_trig_events()
        if len(trig_events) > 0:
            self.get_logger().info('I heard a trig signal')
            print(trig_events)
```
The cd_events seems to decode successfully, but self.decoder.get_ext_trig_events() always returns an empty list. I have confirmed with event_camera_tools that my trig signal is correct.

Currently, one workaround I figured out is to use event_camera_tools to republish the trig events. Then I can receive them using topic '/event_camera/republished_triggers'. But I wonder if we can use event_camera_py  alone to decode the trig events? Any suggestions will be much appreciated. Thanks!

**Comment by berndpfrommer** [comment]
Traveling and didn't take my laptop with me.
I would compile event_camera_py from source and stick print statements into the C++ modules to see if trig events are decided there.
Also please record a very short rosbag with trig events and send as link to it to my email. Thanks

**Comment by NetaPanda** [comment]
Good idea, I am rather new to ROS and not sure if I can handle the C++ modules smoothly, will give it a try. 

Here is the rosbag, with 5Hz square wave trig (i.e., alternative 0 and 1 trigs every 0.1 second):
[https://drive.google.com/drive/folders/1z8hy3-fLydFMUqxWYdFiG0bwDLfHlzkh?usp=sharing](url)

**Comment by NetaPanda** [comment]
After removing the apt-installed package and building `event_camera_py `from source, the issue was unexpectedly resolved. Interestingly, when I reinstalled the apt package to check if it was an apt package problem, the apt package also worked fine. How strange...I suspect that my previously cluttered workspace might have caused some package interference, leading to this bug. And the re-building of `event_camera_py ` happens to solve it.

Thanks again for the reply, looks like I should not put multiple `ros-event-camera `packages under the same workspace src folder. Closing this issue now.
